samples/bluetooth/unicast_audio_server/src/main.c

```diff
@@ -0,0 +1,286 @@
+		printk("%02x", *ptr++);
+	}
+}
+
+static void print_codec(const struct bt_codec *codec)
+{
+	printk("codec 0x%02x cid 0x%04x vid 0x%04x count %u\n",
+	       codec->id, codec->cid, codec->vid, codec->data_count);
+
+	for (size_t i = 0; i < codec->data_count; i++) {
+		printk("data #%zu: type 0x%02x len %u\n",
+		       i, codec->data[i].data.type,
+		       codec->data[i].data.data_len);
+		print_hex(codec->data[i].data.data,
+			  codec->data[i].data.data_len -
+				sizeof(codec->data[i].data.type));
+		printk("\n");
+	}
+
+	for (size_t i = 0; i < codec->meta_count; i++) {
+		printk("meta #%zu: type 0x%02x len %u\n",
+		       i, codec->meta[i].data.type,
+		       codec->meta[i].data.data_len);
+		print_hex(codec->meta[i].data.data,
+			  codec->meta[i].data.data_len -
+				sizeof(codec->meta[i].data.type));
+		printk("\n");
+	}
+}
+
+static void print_qos(struct bt_codec_qos *qos)
+{
+	printk("QoS: dir 0x%02x interval %u framing 0x%02x phy 0x%02x sdu %u "
+	       "rtn %u latency %u pd %u\n",
+	       qos->dir, qos->interval, qos->framing, qos->phy, qos->sdu,
+	       qos->rtn, qos->latency, qos->pd);
+}
+
+static struct bt_audio_chan *lc3_config(struct bt_conn *conn,
+					struct bt_audio_ep *ep,
+					struct bt_audio_capability *cap,
+					struct bt_codec *codec)
+{
+	printk("ASE Codec Config: conn %p ep %p cap %p\n", conn, ep, cap);
+
+	print_codec(codec);
+
+	for (size_t i = 0; i < ARRAY_SIZE(channels); i++) {
+		struct bt_audio_chan *chan = &channels[i];
+
+		if (!chan->conn) {
+			printk("ASE Codec Config chan %p\n", chan);
+			return chan;
+		}
+	}
+
+	printk("No channels available\n");
+
+	return NULL;
+}
+
+static int lc3_reconfig(struct bt_audio_chan *chan,
+			struct bt_audio_capability *cap,
+			struct bt_codec *codec)
+{
+	printk("ASE Codec Reconfig: chan %p cap %p\n", chan, cap);
+
+	print_codec(codec);
+
+	/* We only support one QoS at the moment, reject changes */
+	return -ENOEXEC;
+}
+
+static int lc3_qos(struct bt_audio_chan *chan, struct bt_codec_qos *qos)
+{
+	printk("QoS: chan %p qos %p\n", chan, qos);
+
+	print_qos(qos);
+
+	return 0;
+}
+
+static int lc3_enable(struct bt_audio_chan *chan, uint8_t meta_count,
+		      struct bt_codec_data *meta)
+{
+	printk("Enable: chan %p meta_count %u\n", chan, meta_count);
+
+	return 0;
+}
+
+static int lc3_start(struct bt_audio_chan *chan)
+{
+	printk("Start: chan %p\n", chan);
+
+	return 0;
+}
+
+static int lc3_metadata(struct bt_audio_chan *chan, uint8_t meta_count,
+			struct bt_codec_data *meta)
+{
+	printk("Metadata: chan %p meta_count %u\n", chan, meta_count);
+
+	return 0;
+}
+
+static int lc3_disable(struct bt_audio_chan *chan)
+{
+	printk("Disable: chan %p\n", chan);
+
+	return 0;
+}
+
+static int lc3_stop(struct bt_audio_chan *chan)
+{
+	printk("Stop: chan %p\n", chan);
+
+	return 0;
+}
+
+static int lc3_release(struct bt_audio_chan *chan)
+{
+	printk("Release: chan %p\n", chan);
+
+	return 0;
+}
+
+static struct bt_audio_capability_ops lc3_ops = {
```

Comment:
```suggestion
static const struct bt_audio_capability_ops lc3_ops = {
```

But it should not be possible to do this now, I think `chan.c` does not have constant references.

Reply:
It's a good idea, but as you said, it won't compile without warnings, without changing the audio API first.